src/java/picard/analysis/CollectWgsMetrics.java

```diff
@@ -178,14 +186,21 @@ protected int doWork() {
             // Figure out the coverage while not counting overlapping reads twice, and excluding various things
             final HashSet<String> readNames = new HashSet<String>(info.getRecordAndPositions().size());
             int pileupSize = 0;
+            int pileupSizeForBaseQHetSens = 0;
             for (final SamLocusIterator.RecordAndOffset recs : info.getRecordAndPositions()) {
+                pileupSizeForBaseQHetSens++;
+                if(pileupSizeForBaseQHetSens <= max) {
```

Comment:
This is not your code, but I am not too fond of `max = COVERAGE_CAP`.  When I see `COVERAGE_CAP` I instantly know what it is.  When I see `max`, I don't.